js/basic-cad-text-electron.js

```diff
@@ -101,23 +107,84 @@ function getTextFromData(fontData, fontVariant, text, fontSize, x, y) {
 
 function getTextFromFile(file, text, fontSize, offsetX, offsetY) {
   return getFont(file).then(function(font) {
-    console.log(font)
-    var path = font.getPath(text, 0, 0, fontSize);
+
+    spaceWidth=parseFloat($("#fontspacing").val());
+
+    var unitSwitch = document.getElementById("unitSwitch");
+    if(unitSwitch.checked){
+      var spaceWidth=spaceWidth/25.4;
+      var xpos=16;
+      var ypos=.4;
+    }else{
+      var xpos=0;
+      var ypos=5;
+ 
+    }
+
+    var path = font.getPath(text, ypos, xpos, fontSize);
     var res = '';
+    var xSpace=0;
+    var zFound=[];
+    var ctr=1;
+    var ctr2=1;
+    var x0=0;
+    var x1=0;
+    var x2=0;
+    var xmax1=0;
+    var xmax2=0;
+    var maxEqual=[];
+   
+    // Allow change of spacing in between letters 
+    // run through the letters to find letters that have 2 parts (example A, B, P, R..) 
+    // send it to next for loop so spacing is not added for the second part of th letter
+    for (var i = 0; i < path.commands.length; i++) {
+      
+      var c = path.commands[i];
+         
+      if (c.type == 'M' || c.type == 'L'){
```

Comment:
Something I probably should have pointed out before, but it's not a big deal. The recommended comparison operators in JavaScript are the === (triple equal), not the double equal. There are some nuances with how it compares that boils down to type coercion that can introduce subtle bugs. There may still be times where it's appropriate to use == instead of ===, but it would not be the normal use case. Triple equals should be your default. The existing code base may all use == (haven't checked), so it's not a big deal necessarily, just a best practice I thought I'd mention.

Here's a decent article that explains it: https://www.freecodecamp.org/news/javascript-triple-equals-sign-vs-double-equals-sign-comparison-operators-explained-with-examples/